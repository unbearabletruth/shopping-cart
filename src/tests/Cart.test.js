import { render, screen } from '@testing-library/react';
import Cart from '../components/Cart';
import userEvent from "@testing-library/user-event";

const cart = [
  { image: 'smth', title: "title", author: "author", description: "description", amount: 2, price: 10, id: 1 },
  { image: 'smth1', title: "title1", author: "author1", description: "description1", amount: 1, price: 12, id: 2 },
];

  const increase = jest.fn();
  const decrease = jest.fn();
  const remove = jest.fn();
  const removeAll = jest.fn();
  const change = jest.fn();

test('renders books in cart', () => {
  render(
      <Cart
        cart={cart}
        increase={increase}
        decrease={decrease}
        remove={remove}
        removeAll={removeAll}
        change={change}
      />
  );
  expect(screen.queryAllByTestId('bookInCart').length).toBe(2);
  expect(screen.queryByText('Your cart is empty')).toBeNull();
});

test('removes book from cart', () => {
  render(
    <Cart
      cart={cart}
      increase={increase}
      decrease={decrease}
      remove={remove}
      removeAll={removeAll}
      change={change}
    />
  );

  const button = screen.getByRole('img', {name: "author1"})
  
  userEvent.click(button);

  expect(remove).toHaveBeenCalledTimes(1);
})