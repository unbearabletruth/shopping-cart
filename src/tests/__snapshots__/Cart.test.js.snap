// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart renders proper DOM elements snapshot of cart with books 1`] = `
<div>
  <div
    id="cartWrapper"
  >
    <div
      id="cart"
    >
      <button
        id="removeAllButton"
      >
        remove all
      </button>
      <div
        class="bookInCart"
        data-testid="bookInCart"
      >
        <img
          alt="title"
          class="cartBookImage"
          src="smth"
        />
        <div
          class="titleAndAuthor"
        >
          <p
            class="titleCart"
          >
            title
          </p>
          <p
            class="authorCart"
          >
            author
          </p>
        </div>
        <div
          class="priceBlock"
        >
          <p
            class="priceForOne"
            data-testid="priceForOne"
          >
            One: $
            10.00
          </p>
          <p
            data-testid="priceForMultiple"
          >
            Total: 
            $20.00
          </p>
        </div>
        <div
          class="inputBlock"
        >
          <button
            class="cartInputButton"
            name="1"
          >
            -
          </button>
          <input
            class="inputInShop"
            data-testid="amountInput"
            name="1"
            type="number"
            value="2"
          />
          <button
            class="cartInputButton"
            name="1"
          >
            +
          </button>
        </div>
        <img
          alt="deleteIcon"
          class="deleteImage"
          name="1"
          src="bin.svg"
        />
      </div>
      <div
        class="bookInCart"
        data-testid="bookInCart"
      >
        <img
          alt="title1"
          class="cartBookImage"
          src="smth1"
        />
        <div
          class="titleAndAuthor"
        >
          <p
            class="titleCart"
          >
            title1
          </p>
          <p
            class="authorCart"
          >
            author1
          </p>
        </div>
        <div
          class="priceBlock"
        >
          <p
            class="priceCart"
            data-testid="priceOnlyForOne"
          >
            $
            12.00
          </p>
        </div>
        <div
          class="inputBlock"
        >
          <button
            class="cartInputButton"
            name="2"
          >
            -
          </button>
          <input
            class="inputInShop"
            data-testid="amountInput"
            name="2"
            type="number"
            value="1"
          />
          <button
            class="cartInputButton"
            name="2"
          >
            +
          </button>
        </div>
        <img
          alt="deleteIcon"
          class="deleteImage"
          name="2"
          src="bin.svg"
        />
      </div>
    </div>
    <div
      id="payment"
    >
      <p
        id="totalPriceText"
      >
        Order total
      </p>
      <p
        id="totalPrice"
      >
        $
        32.00
      </p>
      <button
        id="payButton"
      >
        Pay
      </button>
    </div>
  </div>
</div>
`;

exports[`cart renders proper DOM elements snapshot of empty cart 1`] = `
<div>
  <div
    id="emptyCartInfo"
  >
    <p
      id="emptyCartText"
    >
      Your cart is empty
    </p>
    <a
      href="/shop"
      id="cartToShop"
    >
      Fill it in!
    </a>
  </div>
</div>
`;
